### PR TITLE
Remove usage of recarray

### PR DESCRIPTION
**Pull request by timhoffm**
Structured numpy arrays are more fundamental than recarrays and sufficient in all cases.

Superseeds #26664.

### CODE REVIEW

File: galleries/users_explain/quick_start.py [hunk 1]
```diff
@@ -126,10 +126,14 @@
...
-# `numpy.recarray`, or a `pandas.DataFrame`.  Matplotlib allows you to
-# provide the ``data`` keyword argument and generate plots passing the
-# strings corresponding to the *x* and *y* variables.
+# Most methods will also parse a string-indexable object like a *dict*, a
+# `structured numpy array
+# <https://numpy.org/doc/stable/user/basics.rec.html#structured-arrays>`_, or a
+# `pandas.DataFrame`.  Matplotlib allows you to provide the ``data`` keyword argument
+# and generate plots passing the strings corresponding to the *x* and *y* variables.
+#
+# .. _structured numpy array: https://numpy.org/doc/stable/user/basics.rec.html#structured-arrays
```
Comment (ksunden) [inline galleries/users_explain/quick_start.py:135]: This link is in-lined above and causing a flake8 flag

```suggestion
```

Comment (ksunden) [reply galleries/users_explain/quick_start.py:135]: (alternatively we could add this file to the flake8 config ignores)

**Comment by QuLogic** [comment]
> Superseeds #26664.

Note, it's spelled 'supersedes', but also, it's [not one of the valid keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) if that's what you were intending.

**Comment by ksunden** [comment]
> Note, it's spelled 'supersedes'

This always gets me, too, though I tend to think it should be spelled "supercedes" as in "cause the other one to cede to it"... or as opposed to "precede" etc.

File: galleries/examples/misc/keyword_plotting.py [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
+Some data structures, like dict, `structured numpy array
+<https://numpy.org/doc/stable/user/basics.rec.html#structured-arrays>`_
+or `pandas.DataFrame` provide access to labelled data via string index access
+``data[key]``.
 
-Matplotlib allows you to provide such an object with the ``data`` keyword
-argument. If provided, you may generate plots with the strings
-corresponding to these variables.
+For these data types, Matplotlib supports passing the whole datastructure via the
+``data`` keyword argument, and use the string names as plot function parameters, where
```
Comment (QuLogic) [inline galleries/examples/misc/keyword_plotting.py:12]: Match verb tense:
```suggestion
``data`` keyword argument, and using the string names as plot function parameters, where
```

File: galleries/examples/ticks/date_index_formatter.py [hunk 3]
```diff
@@ -32,10 +32,10 @@
 fig.get_layout_engine().set(hspace=0.15)
 
 # First we'll do it the default way, with gaps on weekends
-ax1.plot(r.date, r.adj_close, 'o-')
```
Comment (QuLogic) [inline galleries/examples/ticks/date_index_formatter.py:35]: But the view hasn't been removed from this file?

File: galleries/users_explain/quick_start.py (hunk 1 above)
Comment (timhoffm) [reply galleries/users_explain/quick_start.py]: I don't want to except the whole file from flake8. That's why I've inlined the links. Alternatively, we could use an inline comment `# noqa: E501` on the links, which we generally don't do, but seems ok in this case.

File: galleries/examples/ticks/date_index_formatter.py [hunk 4]
```diff
@@ -25,17 +25,17 @@
...
+ax1.plot(r["date"], r["adj_close"], 'o-')
 
 # Highlight gaps in daily data
-gaps = np.flatnonzero(np.diff(r.date) > np.timedelta64(1, 'D'))
+gaps = np.flatnonzero(np.diff(r["date"]) > np.timedelta64(1, 'D'))
 for gap in r[['date', 'adj_close']][np.stack((gaps, gaps + 1)).T]:
     ax1.plot(gap.date, gap.adj_close, 'w--', lw=2)
```
Comment (QuLogic) [inline galleries/examples/ticks/date_index_formatter.py:40]: This one is a copy of `r` that hasn't been changed.
```suggestion
    ax1.plot(gap['date'], gap['adj_close'], 'w--', lw=2)
```

**Review by QuLogic — APPROVED** [review]

**Pull request merged**